bindings/pyroot/pythonizations/python/ROOT/_pythonization/_roofit/_rooworkspace.py

```diff
@@ -61,8 +61,78 @@ def __setitem__(self, key, value):
                     raise SyntaxError("Invalid syntax")
             else:
                 raise RuntimeError("ERROR importing object named " + key + " another instance with same name already in the workspace and no conflict resolution protocol specified")
+        elif(isinstance(value, dict)):
+            import ROOT
+            import json
+            # Add name attribute to the dictionary, and create the JSON string for the object's JSONTree
+            value["name"] = key
```

Comment:
This `value["name"] = key` is not needed for the variable code branch. Can you put it only in the "is not a variable" part?

Reply:
Done!